### PR TITLE
Python: move configurations out of queries

### PR DESCRIPTION
**Pull request by RasmusWL**
This makes it easy to extend the sources/sinks of the configuration and re-run the query from the query console on LGTM.com.

File location in `semmle.<lang>.security.dataflow.<QueryName>.qll` is matching what we currently do in other languages (JS and C# sampled).

I did not follow the pattern in other languages for wrapping all the code in a module (for example `module CodeInjection`), since I didn't understand the value in doing so -- I would like confirmation from the other teams if we _should_ actually do that, before merging. @hvitved and @asgerf can I hear your opinion on this? (I'd be happy to change our setup to match yours, but I would just like to be convinced it's a good idea first :stuck_out_tongue:)

### CODE REVIEW

**Comment by hvitved** [comment]
> I did not follow the pattern in other languages for wrapping all the code in a module (for example `module CodeInjection`), since I didn't understand the value in doing so -- I would like confirmation from the other teams if we _should_ actually do that, before merging. @hvitved and @asgerf can I hear your opinion on this? (I'd be happy to change our setup to match yours, but I would just like to be convinced it's a good idea first 😛)

AFAIAC there is no need for the extra module layer.

File: python/ql/src/semmle/python/security/dataflow/CodeInjection.qll [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/**
+ * Provides a taint-tracking configuration for reasoning about code injection
```
Comment (yoff) [inline python/ql/src/semmle/python/security/dataflow/CodeInjection.qll:2]:
```suggestion
 * Provides a taint-tracking configuration for detecting code injection
```

File: python/ql/src/semmle/python/security/dataflow/CodeInjection.qll [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ */
+
+import python
+import semmle.python.dataflow.new.DataFlow
+import semmle.python.dataflow.new.TaintTracking
+import semmle.python.Concepts
+import semmle.python.dataflow.new.RemoteFlowSources
+
+/**
+ * A taint-tracking configuration for reasoning about code injection vulnerabilities.
```
Comment (yoff) [inline python/ql/src/semmle/python/security/dataflow/CodeInjection.qll:13]:
```suggestion
 * A taint-tracking configuration for detecting code injection vulnerabilities.
```

File: python/ql/src/semmle/python/security/dataflow/CommandInjection.qll [hunk 3]
```diff
@@ -0,0 +1,51 @@
+/**
+ * Provides a taint-tracking configuration for reasoning about command injection
```
Comment (yoff) [inline python/ql/src/semmle/python/security/dataflow/CommandInjection.qll:2]:
```suggestion
 * Provides a taint-tracking configuration for detecting command injection
```

File: python/ql/src/semmle/python/security/dataflow/CommandInjection.qll [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+ */
+
+import python
+import semmle.python.dataflow.new.DataFlow
+import semmle.python.dataflow.new.TaintTracking
+import semmle.python.Concepts
+import semmle.python.dataflow.new.RemoteFlowSources
+
+/**
+ * A taint-tracking configuration for reasoning about command injection vulnerabilities.
```
Comment (yoff) [inline python/ql/src/semmle/python/security/dataflow/CommandInjection.qll:13]:
```suggestion
 * A taint-tracking configuration for detecting command injection vulnerabilities.
```

File: python/ql/src/semmle/python/security/dataflow/PathInjection.qll [hunk 5]
```diff
@@ -0,0 +1,111 @@
+/**
+ * Provides a taint-tracking configuration for reasoning about path injection
```
Comment (yoff) [inline python/ql/src/semmle/python/security/dataflow/PathInjection.qll:2]:
```suggestion
 * Provides a taint-tracking configuration for detecting path injection
```

File: python/ql/src/semmle/python/security/dataflow/ReflectedXSS.qll [hunk 6]
```diff
@@ -0,0 +1,27 @@
+/**
+ * Provides a taint-tracking configuration for reasoning about reflected server-side
```
Comment (yoff) [inline python/ql/src/semmle/python/security/dataflow/ReflectedXSS.qll:2]:
```suggestion
 * Provides a taint-tracking configuration for detecting reflected server-side
```

File: python/ql/src/semmle/python/security/dataflow/ReflectedXSS.qll [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+import python
+import semmle.python.dataflow.new.DataFlow
+import semmle.python.dataflow.new.TaintTracking
+import semmle.python.Concepts
+import semmle.python.dataflow.new.RemoteFlowSources
+
+/**
+ * A taint-tracking configuration for reasoning about reflected server-side cross-site
```
Comment (yoff) [inline python/ql/src/semmle/python/security/dataflow/ReflectedXSS.qll:13]:
```suggestion
 * A taint-tracking configuration for detecting reflected server-side cross-site
```

File: python/ql/src/semmle/python/security/dataflow/SqlInjection.qll [hunk 8]
```diff
@@ -0,0 +1,21 @@
+/**
+ * Provides a taint-tracking configuration for reasoning about SQL injection
```
Comment (yoff) [inline python/ql/src/semmle/python/security/dataflow/SqlInjection.qll:2]:
```suggestion
 * Provides a taint-tracking configuration for detecting SQL injection
```

File: python/ql/src/semmle/python/security/dataflow/SqlInjection.qll [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+ */
+
+import python
+import semmle.python.dataflow.new.DataFlow
+import semmle.python.dataflow.new.TaintTracking
+import semmle.python.Concepts
+import semmle.python.dataflow.new.RemoteFlowSources
+
+/**
+ * A taint-tracking configuration for reasoning about SQL injection vulnerabilities.
```
Comment (yoff) [inline python/ql/src/semmle/python/security/dataflow/SqlInjection.qll:13]:
```suggestion
 * A taint-tracking configuration for detecting SQL injection vulnerabilities.
```

File: python/ql/src/semmle/python/security/dataflow/UnsafeDeserialization.qll [hunk 10]
```diff
@@ -0,0 +1,27 @@
+/**
+ * Provides a taint-tracking configuration for reasoning about arbitrary code execution
```
Comment (yoff) [inline python/ql/src/semmle/python/security/dataflow/UnsafeDeserialization.qll:2]:
```suggestion
 * Provides a taint-tracking configuration for detecting arbitrary code execution
```

File: python/ql/src/semmle/python/security/dataflow/UnsafeDeserialization.qll [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+import python
+import semmle.python.dataflow.new.DataFlow
+import semmle.python.dataflow.new.TaintTracking
+import semmle.python.Concepts
+import semmle.python.dataflow.new.RemoteFlowSources
+
+/**
+ * A taint-tracking configuration for reasoning about arbitrary code execution
```
Comment (yoff) [inline python/ql/src/semmle/python/security/dataflow/UnsafeDeserialization.qll:13]:
```suggestion
 * A taint-tracking configuration for detecting arbitrary code execution
```

**Review by yoff — CHANGES_REQUESTED** [review]
Thanks for doing this. I suggested a slight change in wording since "reasoning about" seemed a bit to vague and general to me.

**Comment by RasmusWL** [comment]
> Thanks for doing this. I suggested a slight change in wording since "reasoning about" seemed a bit to vague and general to me.

I not against that suggestion, I simply copied the [QLDoc from JS](https://github.com/github/codeql/blob/051c6ca31f8bf18b9890ad3d453ee4d10d3912f1/javascript/ql/src/semmle/javascript/security/dataflow/CodeInjection.qll#L15-L17).

**Review by yoff — APPROVED** [review]
LGTM

**Pull request merged**